device/iot-device-samples/multiplexing-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/MultiplexingSample.java

```diff
@@ -0,0 +1,169 @@
+            }
+            else if (status == IotHubConnectionStatus.DISCONNECTED)
+            {
+                isOpen = false;
+                isDisconnectedRetrying = false;
+            }
+            else if (status == IotHubConnectionStatus.DISCONNECTED_RETRYING)
+            {
+                isDisconnectedRetrying = true;
+                isOpen = false;
+            }
+        }
+    }
+
+    public static int acknowledgedSentMessages = 0;
+    protected static class EventCallback implements IotHubEventCallback
+    {
+        public void execute(IotHubStatusCode status, Object context)
+        {
+            String messageId = (String) context;
+            System.out.println("IoT Hub responded to message "+ messageId  + " with status " + status.name());
+            acknowledgedSentMessages++;
+        }
+    }
+
+    private static final int MULTIPLEXED_DEVICE_CLIENT_COUNT = 3;
+
+    /**
+     * Multiplex devices an IoT Hub using AMQPS / AMQPS_WS
+     *
+     * @param args
+     * args[0] = IoT Hub connection string - Device Client 1
+     * args[1] = IoT Hub connection string - Device Client 2
+     * args[2] = IoT Hub connection string - Device Client 3
+     */
+    public static void main(String[] args)
+            throws IOException, URISyntaxException, InterruptedException {
+        System.out.println("Starting...");
+        System.out.println("Beginning setup.");
+
+        if (args.length != 4)
+        {
+            System.out.format(
+                    "Expected 3 arguments but received: %d.\n"
+                            + "The program should be called with the following args: \n"
+                            + "1. [Device 1 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "2. [Device 2 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "3. [Device 3 connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
+                            + "4. [Protocol]                   - amqps | amqps_ws\n",
+                    args.length);
+            return;
+        }
+
+        String connString1 = args[0];
+        String connString2 = args[1];
+        String connString3 = args[2];
+        IotHubClientProtocol protocol = IotHubClientProtocol.AMQPS;
+        if (args[3].equalsIgnoreCase("amqps_ws"))
+        {
+            protocol = IotHubClientProtocol.AMQPS_WS;
+        }
+
+        MultiplexingClient multiplexingClient = new MultiplexingClient(protocol);
+        DeviceClient[] multiplexedDeviceClients = new DeviceClient[MULTIPLEXED_DEVICE_CLIENT_COUNT];
```

Comment:
As discussed, I think this would work better as a HashMap of `<string deviceId, DeviceClient deviceClient>` and illustrate sending telemetry as devices (and other operations) by looking up a device client by the device Id.

To iterate, you'd have a list of your registered devices, use that to look up your HashMap, and perform the action on the DeviceClient.